### PR TITLE
Add TF support for LocalSsdEncryptionMode flag

### PR DESCRIPTION
**Pull request by halimsam**
Add terraform support for local_ssd_encryption_mode on cluster/nodepool create.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
container: added `local_ssd_encryption_mode` to `node_config` in `google_container_cluster` and `google_container_node_pool`
```

### CODE REVIEW

File: mmv1/third_party/terraform/services/container/node_config.go.tmpl
Comment (rileykarson) [inline mmv1/third_party/terraform/services/container/node_config.go.tmpl:812]: This means that the field has a default value returned by the server even when unset, and won't be returned as `null`. Is that correct? 

`Computed` means that if the field is removed from the config we will preserve the old value instead of revert to null/empty/falsey. If it will only ever be returned if set explicitly, `Computed` can be removed, which is generally preferable. However, if the API may change the default or return non-null defaults in the future (which is breaking for an API but tends to happen regardless) we should keep it.

File: mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.tmpl
Comment (rileykarson) [inline mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.tmpl:4174]: Can we add a second config step with an updated value, since it's mutable? https://googlecloudplatform.github.io/magic-modules/develop/test/test/#add-an-update-test

Only this test is fine- GKE cluster shares code so it'd be redundant.

File: mmv1/third_party/terraform/services/container/node_config.go.tmpl
Comment (rileykarson) [inline mmv1/third_party/terraform/services/container/node_config.go.tmpl:815]: I'd just remove all references to LOCAL_SSD_ENCRYPTION_MODE_UNSPECIFIED unless your API returns it (which I very much doubt- none do that I know of). For Terraform empty/null serves the same purpose.

Comment (halimsam) [reply mmv1/third_party/terraform/services/container/node_config.go.tmpl:815]: Removed.

File: mmv1/third_party/terraform/services/container/node_config.go.tmpl
Comment (halimsam) [reply mmv1/third_party/terraform/services/container/node_config.go.tmpl:812]: This field should only be used (or configured) in boot disk so there shouldn't be any further changes. Will remove `computed`

File: mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.tmpl
Comment (wyardley) [reply mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.tmpl:4174]: side note: if you want to still test the update path in a test, instead of checking for no deletion in the plancheck, you can just not check whether anything's deleted, or you can do something like this instead:
```
				ConfigPlanChecks: resource.ConfigPlanChecks{
					PreApply: []plancheck.PlanCheck{
						plancheck.ExpectResourceAction("xxx.yyy", plancheck.ResourceActionDestroyBeforeCreate),
					},
				},
```

**Review by rileykarson — DISMISSED** [review]

**Comment by rileykarson** [comment]
Running tests before merging, but I don't anticipate any surprises.

**Comment by rileykarson** [comment]
Ah, we're waiting for field availability in the client. Dismissing review until that's ready.

File: mmv1/third_party/terraform/services/container/node_config.go.tmpl
Comment (rileykarson) [reply mmv1/third_party/terraform/services/container/node_config.go.tmpl]: That change seems not to have stuck! I still see it in the ValidateFunc and docs.

Comment (halimsam) [reply mmv1/third_party/terraform/services/container/node_config.go.tmpl]: Apologies, there shouldn't be any references to LOCAL_SSD_ENCRYPTION_MODE_UNSPECIFIED anymore

**Review by rileykarson — APPROVED** [review]

**Pull request merged**
